File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int maxArea = 0;
        int start = 0;
        int end = height.size()-1;

        while(start < end){
            int len = min(height[start], height[end]);
            int d = end-start;
            int curr = len*d;
            maxArea = max(maxArea, curr);
            if(height[start] > height[end]){
                end--;
            }
            else{
                start++;
            }
        }

        return maxArea;
    }
};

File: 11-container-with-most-water/container-with-most-water.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """

        l, r = 0, len(height)-1

        maxA = 0
        while (l < r):
            
            currentH = min(height[l], height[r])
            maxA = max(maxA, (currentH * (r-l)))

            if (height[l] < height[r]):
                l += 1
            
            else:
                r -= 1
        

        return maxA


        